### ISSUE TITLE
stdlib: Add `input` function

### ISSUE THREAD

**Issue by spotandjake** [grain-lang/grain#2074]
In python there is the [input](https://docs.python.org/3/library/functions.html#input) function which makes reading inputs syncronously from the command line super easy. This helps to position python in an especially useful place for learning to program and writing simple interactive programs. It would be great if we could add a function like this, currently this behaviour could be done through `stdin` but exposing a nice user friendly function would greatly help with developer experience.

I think the desired type for this would be `input(prompt="") => string`, if we ever get async capabilities in grain, it would also be great if this was an async function. 

We may want to consider adding some sort of `console` library that contains functions like this and possibly additional `printing` utils like `printTable`.

**Comment by ospencer** [comment]
Requires #1583

**Comment by ospencer** [comment]
I also agree that this should be in its own library, including things like `readLine`, etc.
